File: AI.Core/OperatorsImplimentation/MetaOperators/GetOperatorTypeContactsCount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AI.Core.OperatorsImplimentation.MetaOperators
{
    class GetOperatorContactsCount : ChannelOperator
    {
        public GetOperatorContactsCount(Entity thisEntity) 
            : base(thisEntity)
        {
            EnterContactsCount = 1;
        }

        public override void Action()
        {
            ExitContacts.SetValue(ThisEntity.Operators[TargetOperatorId].OperatorType);
        }

        private ulong TargetOperatorId
        {
            get { return EnterContacts[0].Value; }
        }
    }
}

File: NativeCore/OperatorsImplimentation/MetaOperators/GetOperatorTypeContactsCount.cpp
#include "GetOperatorTypeContactsCount.h"

//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Collections::Generic;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Linq;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Text;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Threading::Tasks;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace MetaOperators
			{

				GetOperatorContactsCount::GetOperatorContactsCount(Entity *thisEntity) : ChannelOperator(thisEntity)
				{
					setEnterContactsCount(1);
				}

				void GetOperatorContactsCount::Action()
				{
					ExitContacts->SetValue(getThisEntity()->getOperators()[getTargetOperatorId()]->OperatorType);
				}

				const unsigned long long &GetOperatorContactsCount::getTargetOperatorId() const
				{
					return EnterContacts[0]->Value;
				}
			}
		}
	}
}
